ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/LogRecordWrapper.java

```diff
@@ -235,7 +236,14 @@ public boolean isRemoved() {
 
     public boolean removeFromList(RecordList rl) {
         if (rl != null && rl.size() > 0 && rec != null) {
-            boolean forgotten = forgetRec || rec.forgetHeuristic();
+            // removing transaction  branch at the participant side
+            boolean isParticipantCleaned = rec.forgetHeuristic();
+            if(!isParticipantCleaned && forgetRec) {
+                if (tsLogger.logger.isDebugEnabled())
+                    tsLogger.logger.debugf("Attempt to forget record '%s' by calling bean removal operation failed, going to call abort", rec);
+                isParticipantCleaned = rec.topLevelAbort() == TwoPhaseOutcome.FINISH_OK;
```

Comment:
@mmusgrov iiuc you would say that the tooling should not call topLevelAbort - is that right?

Reply:
@tomjenkinson Yes that is correct, that is what the email thread is trying to resolve, I believe it shouldn't whereas this PR regards it as okay.